.github/workflows/testing.yaml

```diff
@@ -77,7 +76,7 @@ jobs:
       - name: Setup Python
         uses: actions/setup-python@v5
         with:
-          python-version: 3.9
+          python-version: 3.10
```

Comment:
I think this version number needs to be in quotes, that's why you're getting this test failure on `docs`:
```
Version 3.1 was not found in the local cache
  Error: The version '3.1' with architecture 'x64' was not found for Ubuntu 22.04.
  The list of all available versions can be found here: https://raw.githubusercontent.com/actions/python-versions/main/versions-manifest.json
```